server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java

```diff
@@ -55,11 +56,19 @@ public void execute() {
 
   private void checkProjectKeyConsistency(BatchReport.Metadata reportMetadata) {
     String reportProjectKey = projectKeyFromReport(reportMetadata);
-    checkState(
-      ceTask.getComponentKey().equals(reportProjectKey),
-      "ProjectKey in report (%s) is not consistent with projectKey under which the report as been submitted (%s)",
-      reportProjectKey,
-      ceTask.getComponentKey());
+    if (ceTask.getComponentKey() == null) {
+      throw MessageException.of(format(
```

Comment:
It's a totally legit for a CeTask to not have a component key, either it a task unrelated to a component or to a non existing or not yet existing one